### PR TITLE
add failing dt test - and solve it with max peak duration

### PR DESCRIPTION
**Pull request by jorana**
**What is the problem / what does the code in this PR do**
Add a test that will push us into #397.

**Can you briefly describe how it works?**
Make a big bunch of records that should be sufficient to break strax with integer overflows. To solve the issue, we should solve this test.

**Can you give a minimal working example (or illustrate with a figure)?**
This is the example

### CODE REVIEW

**Comment by jorana** [comment]
We must have been really desperate of tests:
afbeelding

File: tests/test_peak_processing.py [hunk 1]
```diff
@@ -87,3 +86,161 @@ def test_sum_waveform(records, peak_left, peak_length):
...
+    # Make a single big peak to contain all the records
+    def retrun_1(x):
+        """
+        Return 1 for all of the input that can be used as a parameter
+            for the splitting in natural breaks
+        :param x: any type of array
+        :return: ones * len(array)
+        """
+        ret = np.ones(len(x))
+        return ret
```
Comment (WenzDaniel) [inline tests/test_peak_processing.py:148]: Can we use a lambda function instead? This looks a bit odd.

File: tests/test_peak_processing.py [hunk 2]
```diff
@@ -87,3 +86,161 @@ def test_sum_waveform(records, peak_left, peak_length):
...
+    t_max = strax.endtime(r)
+    max_record_repetition_factor = 1_000_000
+    for i in range(max_record_repetition_factor):
+        r_copy = r.copy()
+        r_copy['time'] = r_copy['time'] + t_max * i * gap_factor
+        r_buffer.append(r_copy)
+        if r_copy['time'][-1] - r['time'][0] > 1.5 * magic_overflow_time:
+            # No need to go over and beyond. We should get int-overflow
+            break
+    r = np.concatenate(r_buffer)
```
Comment (WenzDaniel) [inline tests/test_peak_processing.py:168]: This is a bit confusing to me-. Are not you just creating equidistant intervals like: 
```
time = np.arange(time, 1,5* magic_overflow_time + t_max*gap_factor, t_max*gap_factor)
```
which would be much simpler?

File: tests/test_peak_processing.py [hunk 3]
```diff
@@ -87,3 +86,161 @@ def test_sum_waveform(records, peak_left, peak_length):
...
+            split_low=True,
+            filter_wing_width=70,
+            min_area=0,
+            do_iterations=2)
+    except AssertionError as e:
+        if not left_extension + max_duration + right_extension < magic_overflow_time:
+            # Ending up here is the ultimate goal of the tests. This
+            # means we are hitting github.com/AxFoundation/strax/issues/397
+            print(f'Great, the test worked, we are getting the assertion '
+                  f'statement for the int overflow')
```
Comment (WenzDaniel) [inline tests/test_peak_processing.py:232]: Would be very strange if we ever trigger this.

**Review by WenzDaniel — APPROVED** [review]
Just some minor comments, in general it is fine.

File: tests/test_peak_processing.py [hunk 4]
```diff
@@ -87,3 +86,161 @@ def test_sum_waveform(records, peak_left, peak_length):
...
+    :param right_extension: option for strax.find_peaks
+    :param gap_threshold: option for strax.find_peaks
+    :return: None
+    """
+
+    # Set this here, no need to test left and right independently
+    left_extension = 0
+    p = np.zeros(0, dtype=strax.peak_dtype())
+    magic_overflow_time = np.iinfo(p.dtype['dt']).max * p.dtype['data'].shape[0]
+    del p
```
[the remark this answers is not in the document]
Comment (jorana) [reply tests/test_peak_processing.py:137]: because I was lazy and used `p` for peaks again below

**Comment by WenzDaniel** [comment]
One more thing I noticed that your changes are also present in: https://github.com/AxFoundation/strax/pull/403/files. We have to be careful with the commit history I think.

File: tests/test_peak_processing.py (hunk 2 above)
Comment (jorana) [reply tests/test_peak_processing.py:168]: indeed, this is exactly what it is doing but I also want to have some records data, if it were just the time fields I was interested in, you would be right.

File: tests/test_peak_processing.py (hunk 3 above)
Comment (jorana) [reply tests/test_peak_processing.py:232]: Yeah, this was the design of the tests as I first wrote the test and then implemented the solution.

**Comment by jorana** [comment]
>One more thing I noticed that your changes are also present in: https://github.com/AxFoundation/strax/pull/403/files. We have to be careful with the commit history I think.

This is true, it's almost exactly the same so should be relatively easy to resolve the merge-conflicts.

File: tests/test_peak_processing.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (jorana) [reply tests/test_peak_processing.py:137]: let me just change it

File: tests/test_peak_processing.py (hunk 3 above)
Comment (WenzDaniel) [reply tests/test_peak_processing.py:232]: But the data stays the same is not it. It is something like result['data'][:] = r['data']

File: tests/test_peak_processing.py (hunk 2 above)
Comment (WenzDaniel) [reply tests/test_peak_processing.py:168]: But the data stays the same is not it. It is something like result['data'][:] = r['data']

File: tests/test_peak_processing.py (hunk 1 above)
Comment (jorana) [reply tests/test_peak_processing.py:148]: lambda is very non pep8 right? What would your main motivation be for making this a lambda?

Comment (WenzDaniel) [reply tests/test_peak_processing.py:148]: Hmm, I to my understand something like: 
```
function(a, b, c labda x: do_something_simple)
```
would be the default example for a lambda function. Like used in `filter`, `map` etc.   It is the only places where I would use them.

File: tests/test_peak_processing.py (hunk 2 above)
Comment (jorana) [reply tests/test_peak_processing.py]: okay, yeah it is a bit convoluted, I've changed to `np.tile` to do almost exactly the same, thanks for the suggestion

File: tests/test_peak_processing.py (hunk 1 above)
Comment (jorana) [reply tests/test_peak_processing.py:146]: @WenzDaniel , I just stumbled across the pep8 guidelines, and I must say, I would agree with their reasoning:
https://www.python.org/dev/peps/pep-0008/

**Pull request merged**